pulsar-broker/src/main/java/org/apache/pulsar/broker/service/HashRangeExclusiveStickyKeyConsumerSelector.java

```diff
@@ -52,8 +54,23 @@ public HashRangeExclusiveStickyKeyConsumerSelector(int rangeSize) {
     }
 
     @Override
-    public void addConsumer(Consumer consumer) throws BrokerServiceException.ConsumerAssignException {
-        validateKeySharedMeta(consumer);
+    public synchronized CompletableFuture<Void> addConsumer(Consumer consumer) {
+        return validateKeySharedMeta(consumer).thenRun(() -> {
```

Comment:
Nit: a minor optimization could assign `validateKeySharedMeta` to a local variable, and then check if that future is already completed. When it is, we know we had the `synchronized` lock when the validation was done, which would allow us to skip the secondary call to `findConflictingConsumer`. I am not sure how expensive that call is, but I assume it has some cost that adds up with many key shared consumers.

Since it is an optimization, I don't think we should hold up this PR for that.